src/interfaces/IPermissionHook.sol

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.25;
+
+import {PackedUserOperation} from "@eth-infinitism/account-abstraction/interfaces/PackedUserOperation.sol";
+import {IExecutionHook} from "./IExecutionHook.sol";
+
+interface IPermissionHook is IExecutionHook {
```

Comment:
Why do we introduce a new interface here? Can't they use the same `IExecutonHook` as before?

Reply:
I think permission hooks should apply to both functions that provide context - `executeUserOp` and `executeWithAuthorization`. In both cases, the data format provided is different - one would have PackedUserOperation, one would have just msg.data

I considered the alternative (previous approach) of having a single hook, but data could be 2 different formats. Compared to that, this impl doesn't add any new concept since plugin devs already have to consider both cases. But with 2 different hooks formats, plugin devs are able to do bytes4 selector = data[:4] and more importantly, (...) = abi.decode(data[4:], (...))

Agree its not great, but don't see a better way forward - lmk if you have 1 in mind!